stats/src/main/java/com/facebook/stats/mx/JVMStatsExporter.java

```diff
@@ -49,6 +54,34 @@
     .add(SimpleType.DOUBLE)
     .build();
 
+  private static final Set<String> NUMERIC_STRING_TYPES = ImmutableSet.<String>builder()
+    .add(byte.class.getName()).add(Byte.class.getName())
+    .add(short.class.getName()).add(Short.class.getName())
+    .add(int.class.getName()).add(Integer.class.getName())
+    .add(long.class.getName()).add(Long.class.getName())
+    .add(float.class.getName()).add(Float.class.getName())
+    .add(double.class.getName()).add(Double.class.getName())
+    .build();
+
+  /**
+   * Decides whether to include MBean attributes in the Stats object, and, if
+   * so, what to call them.
+   */
+  public interface StatsNameBuilder {
+    /**
+     * Gets the name that should be used in the Stats for an attribute.
+     *
+     * @param bean the ObjectName of the bean where the attribute was found
+     * @param attribute the name of the top-level attribute
+     * @param keys if the bean's top-level attribute was {@link CompositeData},
+     *             the keys used to get from there to the value being named.
+     *
+     * @return the name that should be used, or {@link Optional#none()} if it
+     *         should be elided
+     */
+    Optional<String> name(ObjectName bean, String attribute, String... keys);
```

Comment:
The docs are incorrect; this is not parse **all** the keys, but rather called with one key at a time. Think that's the better API anyway, so maybe update to:

    public interface StatsNameBuilder {
      /**
       * Gets the name that should be used in the Stats for an attribute.
       *
       * @param bean the ObjectName of the bean where the attribute was found
       * @param attribute the name of the top-level attribute
       * @param key if the bean's top-level attribute was {@link CompositeData},
       *            the key used to get from there to the value being named.
       *
       * @return the name that should be used, or {@link Optional#none()} if it
       *         should be elided
       */
      Optional<String> name(ObjectName bean, String attribute, String key);
    }